### PR TITLE
Hrms otp

### PR DESCRIPTION
**Pull request by Sameerthe-Big-O**
Otp as a  middleware on forgot,sign-up,login also some minor changes

### CODE REVIEW

File: config/env.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
-  port: process.env.PORT || 2022,
-  nodeEnv: process.env.NODE_ENV || "development",
-  mongodbUri: process.env.DB_URI || "mongodb://127.0.0.1:27017/boilerplate",
-};
-
-export default environment;
+const environment = {
+  port: process.env.PORT || 2022,
+  nodeEnv: process.env.NODE_ENV || "development",
+  mongodbUri: process.env.DB_URI || "mongodb://127.0.0.1:27017/HRMS",
```
Comment (AbubakarZorrain) [inline config/env.js:4]: use Live MongoDb

File: middlewares/optgeneretor.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    port: 587,
+    auth: {
+        user: 'emerson.ebert@ethereal.email',
+        pass: 'QNFN4vFS1Eg7vgMJCj'
+    }
+  });
+
+  const mailOptions = {
+    from: 'sameer.ijaz100@gmail.com',
+    to: 'emerson.ebert@ethereal.email',
```
Comment (AbubakarZorrain) [inline middlewares/optgeneretor.js:21]: why it is hard coded? Get it from user request.

File: models/attendence.js [hunk 3]
```diff
@@ -1,37 +1,37 @@
...
+        maxlength: 250,
+    },
+    monthly_hours:{
+        type: Number,
+        required: true
+    },
+},
+// { timestamps: true },
+);
+export default mongoose.model("Attendence", schema);
```
Comment (AbubakarZorrain) [inline models/attendence.js:37]: Does Attendance schema necessary for otp ?

File: routes/attendence/controllers.js [hunk 4]
```diff
@@ -1,56 +1,56 @@
...
+    // }
+  },
+  delete: async (req, res) => {
+    try {
+        const addResponse = await AttendenceService.delete(req.params.id);   
+        return httpResponse.SUCCESS(res, addResponse.data);  
+    } catch (error) {
+        return httpResponse.NOT_FOUND(res, error);
+    }
+    // if (addResponse.message === "success") {
```
Comment (AbubakarZorrain) [inline routes/attendence/controllers.js:46]: Why comments are there?

**Review by AbubakarZorrain — CHANGES_REQUESTED** [review]

**Comment by AbubakarZorrain** [comment]
above code is not much related to otp section. what are you doing for otp generation? Literally 40 files changed....

File: config/env.js (hunk 1 above)
Comment (Sameerthe-Big-O) [reply config/env.js:4]: Aoa,sir according to the documentation we've to use MongoDB compass but as you've mentioned using MongoDB live you're referring to the Atlas version ??

File: middlewares/optgeneretor.js (hunk 2 above)
Comment (Sameerthe-Big-O) [reply middlewares/optgeneretor.js:21]: okay sir

File: models/attendence.js (hunk 3 above)
Comment (Sameerthe-Big-O) [reply models/attendence.js:37]: Sir i think it depends on the policy of any organization they're strict on this or not like if they want to make sure that the employee should be physically there then **yes** the OTP will be sent either on the device that's in the organization, at the same time if we're talking that without physical attendance then I really don't think we need to send

File: routes/attendence/controllers.js (hunk 4 above)
Comment (Sameerthe-Big-O) [reply routes/attendence/controllers.js:46]: this was my mistake i've removed these comments

**Comment by Sameerthe-Big-O** [comment]
closing the pr

**Pull request closed without merging**